Index: tvtime/src/tvtime-command.c
===================================================================
--- tvtime.orig/src/tvtime-command.c	2014-04-25 17:27:37.935860003 +0200
+++ tvtime/src/tvtime-command.c	2014-04-25 17:27:37.931859906 +0200
@@ -39,6 +39,9 @@
 
 int main( int argc, char **argv )
 {
+    if( ! getenv( "HOME" ) ) { fprintf( stderr, "this program needs HOME\n" ); exit( 1 ); }
+    if( strlen( getenv( "HOME" ) ) > 230 ) { fprintf( stderr, "HOME is too long\n" ); exit( 1 ); }
+
     int nc = tvtime_num_commands();
     config_t *cfg;
     FILE *fifo;
Index: tvtime/src/tvtime-configure.c
===================================================================
--- tvtime.orig/src/tvtime-configure.c	2014-04-25 17:27:37.935860003 +0200
+++ tvtime/src/tvtime-configure.c	2014-04-25 17:36:42.644717220 +0200
@@ -19,6 +19,8 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <string.h>
+#include <stdlib.h>
 #include <fcntl.h>
 #include <unistd.h>
 #ifdef HAVE_CONFIG_H
@@ -35,6 +37,9 @@
 
 int main( int argc, char **argv )
 {
+    if( ! getenv( "HOME" ) ) { fprintf( stderr, "this program needs HOME\n" ); exit( 1 ); }
+    if( strlen( getenv( "HOME" ) ) > 230 ) { fprintf( stderr, "HOME is too long\n" ); exit( 1 ); }
+
     config_t *cfg;
 
     /*
Index: tvtime/src/tvtime-scanner.c
===================================================================
--- tvtime.orig/src/tvtime-scanner.c	2014-04-25 17:27:37.935860003 +0200
+++ tvtime/src/tvtime-scanner.c	2014-04-25 17:36:20.736208769 +0200
@@ -24,6 +24,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <string.h>
+#include <stdlib.h>
 #include <fcntl.h>
 #include <unistd.h>
 #ifdef HAVE_CONFIG_H
@@ -42,6 +43,9 @@
 
 int main( int argc, char **argv )
 {
+    if( ! getenv( "HOME" ) ) { fprintf( stderr, "this program needs HOME\n" ); exit( 1 ); }
+    if( strlen( getenv( "HOME" ) ) > 230 ) { fprintf( stderr, "HOME is too long\n" ); exit( 1 ); }
+
     config_t *cfg;
     station_mgr_t *stationmgr = 0;
     videoinput_t *vidin;
Index: tvtime/src/tvtime.c
===================================================================
--- tvtime.orig/src/tvtime.c	2014-04-25 17:27:37.935860003 +0200
+++ tvtime/src/tvtime.c	2014-04-25 17:27:37.935860003 +0200
@@ -2643,6 +2643,9 @@
 
 int main( int argc, char **argv )
 {
+    if( ! getenv( "HOME" ) ) { fprintf( stderr, "this program needs HOME\n" ); exit( 1 ); }
+    if( strlen( getenv( "HOME" ) ) > 230 ) { fprintf( stderr, "HOME is too long\n" ); exit( 1 ); }
+
     rtctimer_t *rtctimer = 0;
     int read_stdin = 1;
     int result = 0;
